File: Daily/2023/35_searchInsertPosition.py
from typing import *

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        size = len(nums)
        left, right = 0, size - 1

        while left < right:
            mid = left + (right - left) // 2 + 1
            if nums[mid] > target:
                right = mid - 1
            else:
                left = mid
            
        return left + 1 if target > nums[left] else left

File: Categories/Binary_Search/35_searchInsertPosition.cs
public class Solution {
    public int SearchInsert(int[] nums, int target) {
        int n = nums.Length;
        int left = 0, right = n - 1;

        while (left < right) {
            int mid = (left + right) / 2;
            if (nums[mid] < target) {
                left = mid + 1;
            }
            else {
                right = mid;
            }
        }

        return nums[left] < target ? left + 1: left;
    }
}